model/src/main/java/de/danoeh/antennapod/model/feed/FeedPreferences.java

```diff
@@ -168,12 +191,30 @@ public void setFeedID(long feedID) {
         this.feedID = feedID;
     }
 
-    public boolean getAutoDownload() {
-        return autoDownload;
+    public boolean getAutoDownload(boolean globalDefault) {
+        switch (this.autoDownload) {
+            case ENABLED -> {
+                return true;
+            }
+            case DISABLED -> {
+                return false;
+            }
+            default -> {
+                return globalDefault;
+            }
+        }
+    }
+
+    public AutoDownloadSetting getAutoDownloadRaw() {
```

Comment:
For other settings, we basically only have the raw method (not called raw) and handle the global default outside this class

Reply:
I would like to keep this (but maybe rename it). My reasoning is that often i don't really care what the global value is, just the end-result. (I would have moved the `UserPreferences.defaultAutodownloadState()` also inside this method, but that would add a dependency to that module)

This makes the abstraction a bit easier since i don't need to reimplement the calculation everywhere.